app/src/sharedTest/java/org/oppia/android/app/settings/profile/ProfileRenameFragmentTest.kt

```diff
@@ -0,0 +1,481 @@
+      Espresso.onView(ViewMatchers.withId(R.id.profile_rename_input))
+        .check(ViewAssertions.matches(TextInputAction.hasNoErrorText()))
+    }
+  }
+
+  @Test
+  fun testProfileRenameFragment_inputName_changeConfiguration_checkNameIsDisplayed() {
+    ActivityScenario.launch<ProfileRenameActivity>(
+      ProfileRenameActivity.createProfileRenameActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      Espresso.onView(
+        CoreMatchers.allOf(
+          ViewMatchers.withId(R.id.profile_rename_input_edit_text),
+          ViewMatchers.isDescendantOfA(ViewMatchers.withId(R.id.profile_rename_input))
+        )
+      ).perform(
+        editTextInputAction.appendText("test"),
+        ViewActions.closeSoftKeyboard()
+      )
+      testCoroutineDispatchers.runCurrent()
+      Espresso.onView(ViewMatchers.isRoot()).perform(OrientationChangeAction.orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      Espresso.onView(
+        CoreMatchers.allOf(
+          ViewMatchers.withId(R.id.profile_rename_input_edit_text),
+          ViewMatchers.isDescendantOfA(ViewMatchers.withId(R.id.profile_rename_input))
+        )
+      ).check(
+        ViewAssertions.matches(
+          ViewMatchers.withText("test")
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testProfileRenameFragment_inputOldName_clickSave_changeConfiguration_errorIsVisible() {
+    ActivityScenario.launch<ProfileRenameActivity>(
+      ProfileRenameActivity.createProfileRenameActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      Espresso.onView(
+        CoreMatchers.allOf(
+          ViewMatchers.withId(R.id.profile_rename_input_edit_text),
+          ViewMatchers.isDescendantOfA(ViewMatchers.withId(R.id.profile_rename_input))
+        )
+      ).perform(
+        editTextInputAction.appendText("Admin"),
+        ViewActions.closeSoftKeyboard()
+      )
+      testCoroutineDispatchers.runCurrent()
+      Espresso.onView(ViewMatchers.withId(R.id.profile_rename_save_button))
+        .perform(ViewActions.click())
+      testCoroutineDispatchers.runCurrent()
+      Espresso.onView(ViewMatchers.isRoot()).perform(OrientationChangeAction.orientationLandscape())
+      testCoroutineDispatchers.runCurrent()
+      Espresso.onView(ViewMatchers.withId(R.id.profile_rename_input))
+        .check(
+          ViewAssertions.matches(
+            TextInputAction.hasErrorText(
+              context.resources.getString(R.string.add_profile_error_name_not_unique)
+            )
+          )
+        )
+    }
+  }
+
+  @Test
+  fun testProfileRenameFragment_clickSave_changeConfiguration_saveButtonIsNotClickable() {
+    ActivityScenario.launch<ProfileRenameActivity>(
+      ProfileRenameActivity.createProfileRenameActivity(
+        context = context,
+        profileId = 1
+      )
+    ).use {
+      Espresso.onView(ViewMatchers.withId(R.id.profile_rename_save_button))
```

Comment:
Please use direct imports everywhere
Example
`onView(withId(R.id.profile_rename_save_button)).check(matches(not(isClickable())))`

This makes code more readable.

Reply:
fixed